pkg/kudoctl/util/repo/repo_framework.go

```diff
@@ -0,0 +1,220 @@
+	if err != nil {
+		return nil, errors.Wrap(err, "parsing config url")
+	}
+	parsedURL.Path = parsedURL.Path + "/" + bundleName + ".tgz"
+
+	fileURL = parsedURL.String()
+
+	resp, err := r.Client.Get(fileURL)
+	if err != nil {
+		return nil, errors.Wrap(err, "getting file url")
+	}
+
+	bundle, err := untar(resp)
+
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed unpacking %s", bundleName)
+	}
+
+	return bundle, nil
+}
+
+// GetFrameworkVersion gets the proper Framework version of a given Framework
+func (r *FrameworkRepository) GetFrameworkVersion(name, path string) (*v1alpha1.FrameworkVersion, error) {
+	frameworkVersionPath := path + "/" + name + "-frameworkversion.yaml"
+	frameworkVersionYamlFile, err := os.Open(frameworkVersionPath)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed opening frameworkversion file %s", frameworkVersionPath)
+	}
+
+	frameworkVersionByteValue, err := ioutil.ReadAll(frameworkVersionYamlFile)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed reading frameworkversion file %s", frameworkVersionPath)
+	}
+
+	var fv v1alpha1.FrameworkVersion
+	err = yaml.Unmarshal(frameworkVersionByteValue, &fv)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unmarshalling %s-frameworkversion.yaml content", name)
+	}
+	return &fv, nil
+}
+
+// GetFrameworkVersionDependencies returns a slice of strings that contains the names of all dependency Frameworks
+func (r *FrameworkRepository) GetFrameworkVersionDependencies(name string, fv *v1alpha1.FrameworkVersion) ([]string, error) {
+	var dependencyFrameworks []string
+	if fv.Spec.Dependencies != nil {
+		for _, v := range fv.Spec.Dependencies {
+			dependencyFrameworks = append(dependencyFrameworks, v.Name)
+		}
+	}
+	return dependencyFrameworks, nil
+}
+
+func untar(r io.Reader) (*FrameworkBundle, error) {
+
+	gzr, err := gzip.NewReader(r)
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		err := gzr.Close()
+		if err != nil {
+			fmt.Printf("Error when closing gzip reader %s", err)
+		}
+	}()
+
+	tr := tar.NewReader(gzr)
+
+	result := &FrameworkBundle{}
+	for {
+		header, err := tr.Next()
+
+		switch {
+
+		// if no more files are found return
+		case err == io.EOF:
+			return result, nil
+
+		// return any other error
+		case err != nil:
+			return nil, err
+
+		// if the header is nil, just skip it (not sure how this happens)
+		case header == nil:
+			continue
+		}
+
+		// check the file type
+		switch header.Typeflag {
+
+		case tar.TypeDir:
+			// we don't handle folders right now, the structure is flat
+
+		// if it's a file create it
+		case tar.TypeReg:
+			bytes, err := ioutil.ReadAll(tr)
+			if err != nil {
+				return nil, errors.Wrapf(err, "while reading file from bundle tarball %s", header.Name)
+			}
+
+			switch {
+			case isFrameworkFile(header.Name):
+				var f v1alpha1.Framework
+				if err = yaml.Unmarshal(bytes, &f); err != nil {
+					return nil, errors.Wrapf(err, "unmarshalling %s-framework.yaml content", header.Name)
```

Comment:
We don't need `%s-framework.yaml` as `header.Name` contains it already. See a possible error message

```
Error: could not install framework(s): failed to download bundle: failed unpacking test-0.1.0: unmarshalling ./._test-framework.yaml-framework.yaml content: error converting YAML to JSON: yaml: control characters are not allowed
```

Instead just use:
```
return nil, errors.Wrapf(err, "unmarshalling %s content", header.Name)
```